node/directives/withCurrentProfile.ts

```diff
@@ -200,10 +200,18 @@ async function checkUserAccount(
     logger.warn(logData)
   }
 
+  const isUserCallCenterOperator =
+    'id' in tokenUser
+      ? await isValidCallcenterOperator(context, tokenUser.user)
+      : false
+
   if (
     tokenUser &&
     'id' in tokenUser &&
-    !(tokenUser.account === account && tokenUser.user === userProfile.email)
+    !(
+      tokenUser.account === account &&
+      (isUserCallCenterOperator || tokenUser.user === userProfile.email)
```

Comment:
Do you really need to validate the email? Since you already validated the account the problem with cross-account is already solved, trying to avoid unnecessary calls to the LM service. If you really need to check the email perhaps You could only see if the user has an `adminUserAuthToken` it means that they are accessing via `account.myvtex.com` rather than the default domain which is accessed by the store customers

Reply:
So, you mean i should only check if:
`adminUserAuthToken || tokenUser.user === userProfile.email` ?